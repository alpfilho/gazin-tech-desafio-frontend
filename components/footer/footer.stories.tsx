import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Footer } from './footer';

export default {
	title: 'Components/Footer',
	component: Footer,
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof Footer>;

export const Default: ComponentStory<typeof Footer> = (args) => <Footer {...args} />;

Default.args = {};
