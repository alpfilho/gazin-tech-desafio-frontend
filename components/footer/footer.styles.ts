import styled from 'styled-components';

import { Container, Content } from 'components/contentContainer';

export const FooterContainer = styled.footer`
	width: 100%;
	color: ${(props) => props.theme.text};
	background-color: ${(props) => props.theme.background};
	transition: color 250ms linear, background-color 250ms linear;
	box-shadow: 0 -1em 10em rgba(0, 0, 0, 0.33);
`;

export const LogosContainer = styled.div`
	width: 100%;
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: flex-end;

	@media screen and (max-width: 992px) {
		flex-direction: column;
		align-items: center;
		justify-content: center;
	}
`;

export const LogoBox = styled.a`
	width: 50%;
	max-width: 16em;

	display: flex;
	justify-content: center;
	align-items: center;
	padding-top: 1.5em;
`;

export const TmdbLogoBox = styled(LogoBox)`
	flex-direction: column;

	font-family: 'Roboto', sans-serif;
	font-weight: 400;
	text-align: center;
	padding-top: 1.5em;

	span {
		transition: color 250ms linear;
		color: ${(props) => props.theme.text};
		font-size: 1em;
		padding-bottom: 0.5em;
	}
`;

export const SignatureContainer = styled.div`
	margin-top: 2em;
	width: 100%;
	color: ${(props) => props.theme.text};
	background-color: ${(props) => props.theme.background};
	box-shadow: -0.2em 0 2em rgba(0, 0, 0, 0.33);
	padding-top: 1em;
	padding-bottom: 1em;
	transition: color 250ms linear, background-color 250ms linear;

	${Container} {
		${Content} {
			display: flex;
			justify-content: center;
			align-items: center;
			white-space: pre;

			a {
				transition: color 250ms linear;
				color: ${(props) => props.theme.text};
			}
		}
	}
`;
