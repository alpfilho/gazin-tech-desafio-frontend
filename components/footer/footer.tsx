import React from 'react';
import Link from 'next/link';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import { useTheme } from 'styled-components';

import { ContentContainer } from 'components/contentContainer';
import { TmdbLogo } from 'components/tmdbLogo';
import { Logo } from 'components/logo';

import {
	FooterContainer,
	LogoBox,
	LogosContainer,
	SignatureContainer,
	TmdbLogoBox
} from './footer.styles';

export const Footer: React.FC = () => {
	const theme = useTheme();

	return (
		<FooterContainer className="no-select">
			<ContentContainer>
				<LogosContainer>
					<Link href="/" passHref>
						<LogoBox>
							<Logo color={theme.text} />
						</LogoBox>
					</Link>
					<TmdbLogoBox href="https://www.themoviedb.org/" target="_blank" rel="noreferrer">
						<span>Dados fornecidos por:</span>
						<TmdbLogo type="alt-short" />
					</TmdbLogoBox>
				</LogosContainer>
			</ContentContainer>
			<SignatureContainer>
				<ContentContainer>
					Desenvolvido com <FontAwesomeIcon icon={faHeart} color="#ff2323" /> por{' '}
					<a href="https://github.com/alpfilho" target="_blank" rel="noreferrer">
						Alfredo Peres
					</a>
				</ContentContainer>
			</SignatureContainer>
		</FooterContainer>
	);
};
