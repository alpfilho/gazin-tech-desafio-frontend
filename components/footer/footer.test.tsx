import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './footer.stories';

const { Default } = composeStories(stories);

describe('Footer', () => {
	it('render a footer', () => {
		render(<Default />);

		const devName = screen.getByText('Alfredo Peres');
		expect(devName).toBeInTheDocument();
	});

	it('footer render unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
