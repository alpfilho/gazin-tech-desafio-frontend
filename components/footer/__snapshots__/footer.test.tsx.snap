// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer footer render unchanged 1`] = `
<div>
  <footer
    class="FooterContainer-sc-yclftp-0 eRQyaw no-select"
  >
    <div
      class="Container-sc-yq2cgn-0 leaVgY"
    >
      <div
        class="Content-sc-yq2cgn-1 etbrxE"
      >
        <div
          class="LogosContainer-sc-yclftp-1 lcKkez"
        >
          <a
            class="LogoBox-sc-yclftp-2 brDkyN"
            href="/"
          >
            <svg
              class="LogoContainer-sc-1qw03iv-0 leVVGK"
              data-testid="gazin-logo"
              fill="none"
              viewBox="0 0 909 111"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M71.64 85.9h58.3l31.6-55.97h-44.81l-8.12 14.2 13.53-.11-15.1 26.28s-10.83-.22-11.64-.15c-.81.08-7.79-.61-9.27-1-10.45-2.76-13.21-8.17-14.15-12.65-1.7-8.07 2.65-14.79 2.65-14.79C80.58 31.47 91.13 24.75 93.97 23c9.8-6.07 20.1-7.45 24.44-8.04l1.19-.16c3.3-.5 9.55-.56 9.55-.56h231.52L368.64 0H137.29c-21.72 1.2-32.42 2.97-35.02 3.4l-.23.04C81 6.87 65.67 14.96 63.06 16.43 38.69 30.1 38.93 47.79 38.98 50.98v.11c.26 21.02 19.64 30.7 28.46 34.02 2.6.98 4.2.8 4.2.8ZM160.1 111c-6.1-3.88-10.82-7.54-13.29-14.54H8.2L0 111h160.1Z"
                fill="#282424"
              />
              <path
                d="m257.5 36.38 9.12-16.14h-7.49s-16.38-.13-23.28.36c-24.6 1.75-50.89 8.77-69.98 25.78l-.09.07-.18.17-.09.07c-1.92 1.66-8.08 6.98-13.3 17.47-1.03 2.08-4.37 10.7.08 21.5l.01.03.04.08c.64 1.58 4.48 11.06 19.64 21.95l13.22-14.4s-5.75-4.29-7.33-10.05a16.2 16.2 0 0 1 1.44-10.7l.24-.48c.45-.9.9-1.8 1.48-2.6l28.07-.01-9.25 16.42h29.8l24.5-43.53h-29.79l-5.93 10.53-27.07-.02s5.82-10.96 28.16-15.16c1.97-.37 10-1.34 16.99-1.34h21Zm9.3-.31 8.5-15.83h81.96l-63.51 49.28h35.77l-9.23 16.39H238.3l64.06-49.84H266.8Zm71.88 50.22 36.6-65.68h36.46L374.99 86.3h-36.3Zm39.63-72.09h36.28l8.1-14.2h-36.33l-8.05 14.18ZM434.56 0l-8.02 14.23H530.9L535.8 4h25l-4.9 10.23h23L583.8 4h25l-4.9 10.23h23L631.8 4h25l-4.9 10.23h23L679.8 4h25l-4.9 10.23h23L727.8 4h25l-4.9 10.23h23L775.8 4h25l-4.9 10.23h23L823.8 4h25l-4.9 10.23h23L871.8 4h25l-4.9 10.23h9.1l8-14.22H434.56Z"
                fill="#282424"
              />
              <path
                d="m386.2 85.9 36.95-65.66h36.05l7.07 21.92 12.35-21.92h34L469.37 97h12.53l-4.52 9.64h25L506.9 97h23l-4.52 9.64h25L554.9 97h23l-4.52 9.64h25L602.9 97h23l-4.52 9.64h25L650.9 97h23l-4.52 9.64h25L698.9 97h23l-4.52 9.64h25L746.9 97h23l-4.52 9.64h25L794.9 97h23l-4.52 9.64h25L842.9 97h17.59l-8 14H185.73l8.23-14.54H435.7l16.06-28.52-9.27-25.84-24.66 43.8H386.2Z"
                fill="#282424"
              />
              <path
                d="M827.77 88c-11.83 0-20.83-1.57-27-4.72-6.08-3.15-8.23-7.75-6.45-13.8.18-.6.53-1.51 1.07-2.72h32.21c-.82 1.87-.82 3.26 0 4.17.92.91 2.7 1.36 5.33 1.36 1.64 0 3.09-.27 4.34-.81 1.25-.55 2.03-1.33 2.33-2.36.43-1.46-.3-2.67-2.2-3.63-1.81-.97-4.89-2.12-9.23-3.45a102.24 102.24 0 0 1-11.6-4.18 18.43 18.43 0 0 1-7-5.45c-1.78-2.3-2.16-5.17-1.14-8.62 1.33-4.54 4.16-8.44 8.49-11.71 4.44-3.33 9.95-5.84 16.55-7.54A86.8 86.8 0 0 1 855.1 22c7 0 13.02.73 18.05 2.18 5.13 1.45 8.83 3.57 11.1 6.35 2.38 2.73 3.02 5.97 1.92 9.72-.14.48-.48 1.33-1.02 2.54h-32.62c.4-.73.63-1.21.7-1.45.34-1.15.1-2.03-.73-2.64-.8-.66-2.07-1-3.8-1-1.54 0-2.9.3-4.09.91-1.18.6-1.9 1.4-2.2 2.36-.38 1.34.32 2.52 2.1 3.54a70.9 70.9 0 0 0 8.9 3.64 102.4 102.4 0 0 1 11.47 4.63 18.43 18.43 0 0 1 7.16 5.8c1.76 2.37 2.14 5.27 1.12 8.72-1.15 3.94-3.79 7.48-7.9 10.62-4.02 3.15-9.25 5.63-15.7 7.45-6.43 1.75-13.7 2.63-21.79 2.63ZM582.23 39.07l8.93-15.98h-68.24l-35.96 64.27h30.3l13.54-24.23h27.3l8.55-15.17h-27.3l4.94-8.9h37.94Zm8.07 48.29 35.97-64.27h-30.3L560 87.36h30.3Zm47.1-15.25h29.21l-8.58 15.25h-59.5l35.96-64.27h30.3L637.4 72.1Zm140.85 15.25 35.97-64.27h-36.99l-42.36 38.67.32-38.67h-37.12l-35.96 64.27h30.3l17.94-32.22-3.07 32.22h25.94l32.73-31.95-17.86 31.95h30.16Z"
                fill="#282424"
              />
            </svg>
          </a>
          <a
            class="LogoBox-sc-yclftp-2 TmdbLogoBox-sc-yclftp-3 brDkyN hSXmsT"
            href="https://www.themoviedb.org/"
            rel="noreferrer"
            target="_blank"
          >
            <span>
              Dados fornecidos por:
            </span>
            <svg
              data-testid="tmdb-logo-alt-short"
              viewBox="0 0 273.42 35.52"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="imdb-logo-gradient-2"
                  x2="273.42"
                  y1="17.76"
                  y2="17.76"
                >
                  <stop
                    offset="0"
                    stop-color="#90cea1"
                  />
                  <stop
                    offset=".56"
                    stop-color="#3cbec9"
                  />
                  <stop
                    offset="1"
                    stop-color="#00b3e5"
                  />
                </lineargradient>
              </defs>
              <g
                data-name="Layer 2"
              >
                <path
                  d="M191.85 35.37h63.9a17.67 17.67 0 0 0 17.67-17.67A17.67 17.67 0 0 0 255.75 0h-63.9a17.67 17.67 0 0 0-17.67 17.7 17.67 17.67 0 0 0 17.67 17.67Zm-181.75.05h7.8V6.92H28V0H0v6.9h10.1Zm28.1 0H46V8.25h.1l8.95 27.15h6L70.3 8.25h.1V35.4h7.8V0H66.45l-8.2 23.1h-.1L50 0H38.2ZM89.14.12h11.7a33.56 33.56 0 0 1 8.08 1 18.52 18.52 0 0 1 6.67 3.08 15.09 15.09 0 0 1 4.53 5.52 18.5 18.5 0 0 1 1.67 8.25 16.91 16.91 0 0 1-1.62 7.58 16.3 16.3 0 0 1-4.38 5.5 19.24 19.24 0 0 1-6.35 3.37 24.53 24.53 0 0 1-7.55 1.15H89.14Zm7.8 28.2h4a21.66 21.66 0 0 0 5-.55A10.58 10.58 0 0 0 110 26a8.73 8.73 0 0 0 2.68-3.35 11.9 11.9 0 0 0 1-5.08 9.87 9.87 0 0 0-1-4.52 9.17 9.17 0 0 0-2.63-3.18A11.61 11.61 0 0 0 106.22 8a17.06 17.06 0 0 0-4.68-.63h-4.6ZM133.09.12h13.2a32.87 32.87 0 0 1 4.63.33 12.66 12.66 0 0 1 4.17 1.3 7.94 7.94 0 0 1 3 2.72 8.34 8.34 0 0 1 1.15 4.65 7.48 7.48 0 0 1-1.67 5 9.13 9.13 0 0 1-4.43 2.82V17a10.28 10.28 0 0 1 3.18 1 8.51 8.51 0 0 1 2.45 1.85 7.79 7.79 0 0 1 1.57 2.62 9.16 9.16 0 0 1 .55 3.2 8.52 8.52 0 0 1-1.2 4.68 9.32 9.32 0 0 1-3.1 3 13.38 13.38 0 0 1-4.27 1.65 22.5 22.5 0 0 1-4.73.5h-14.5Zm7.8 14.15h5.65a7.65 7.65 0 0 0 1.78-.2 4.78 4.78 0 0 0 1.57-.65 3.43 3.43 0 0 0 1.13-1.2 3.63 3.63 0 0 0 .42-1.8A3.3 3.3 0 0 0 151 8.6a3.42 3.42 0 0 0-1.23-1.13A6.07 6.07 0 0 0 148 6.9a9.9 9.9 0 0 0-1.85-.18h-5.3Zm0 14.65h7a8.27 8.27 0 0 0 1.83-.2 4.67 4.67 0 0 0 1.67-.7 3.93 3.93 0 0 0 1.23-1.3 3.8 3.8 0 0 0 .47-1.95 3.16 3.16 0 0 0-.62-2 4 4 0 0 0-1.58-1.18 8.23 8.23 0 0 0-2-.55 15.12 15.12 0 0 0-2.05-.15h-5.9Z"
                  data-name="Layer 1"
                  fill="url(#imdb-logo-gradient-2)"
                />
              </g>
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="SignatureContainer-sc-yclftp-4 ddQRLi"
    >
      <div
        class="Container-sc-yq2cgn-0 leaVgY"
      >
        <div
          class="Content-sc-yq2cgn-1 etbrxE"
        >
          Desenvolvido com 
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart "
            color="#ff2323"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
              fill="currentColor"
            />
          </svg>
           por
           
          <a
            href="https://github.com/alpfilho"
            rel="noreferrer"
            target="_blank"
          >
            Alfredo Peres
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
